unlock-app/src/components/interface/locks/Settings/forms/UpdateReferralFee.tsx

```diff
@@ -105,19 +114,78 @@ export const UpdateReferralFee = ({
             errors?.referralFeePercentage &&
             'This field accept percentage value between 0 and 100.'
           }
-          disabled={disabledInput || !referralFee}
+          disabled={isDisabledReferrerInput || !isReferralFeeEnabled}
         />
       </div>
+
+      <div className="grid gap-2">
+        <ToggleSwitch
+          title="Referrer address"
+          enabled={isReferralAddressEnabled}
+          disabled={isDisabledReferrerInput}
+          setEnabled={(enabled: boolean) => {
+            setIsReferralAddressEnabled(enabled)
+          }}
+        />
+        <Input
+          type="text"
+          {...register('referralAddress', {
+            validate: (value) =>
+              ethers.utils.isAddress(value.toLowerCase()) || 'Invalid address',
+            required: {
+              value: true,
+              message: 'This field is required.',
+            },
+          })}
+          error={errors?.referralAddress?.message}
+          disabled={isDisabledReferrerInput || !isReferralAddressEnabled}
+        />
+      </div>
+
       {isManager && (
         <Button
           className="w-full md:w-1/3"
           type="submit"
-          disabled={disabledInput}
+          disabled={isDisabledReferrerInput}
           loading={setReferrerFeeMutation.isLoading}
         >
           Apply
         </Button>
       )}
+
+      <div className="flex flex-col mt-5 items-start">
+        <Card.Title>Referrers ({referralFees.length})</Card.Title>
```

Comment:
You will have a complete overview of the referrers set per address on the lock you created ✅.